src/stream.js

```diff
@@ -21,8 +26,63 @@ module.exports.handler = async (event) => {
 	delete(groupedRecords['deleteMe']); // delete bad events
 
 	for (const records of Object.entries(groupedRecords)) {
-		const latestRecord = records.pop();
+		logger.debug({records}, 'Records');
+		const latestRecord = records.pop().pop();
+		// Tag for a transient record that should be ignored.
+		if (latestRecord.date === '_current') {
+			continue;
+		}
 		logger.debug({latestRecord}, 'Event received');
-	}
 
+		const responses = latestRecord.users && Object.keys(latestRecord.users).map(key => {
+			return {
+				name: `<@${key}>`,
+				answer: latestRecord.users[key] && latestRecord.users[key].answer  === 'fact'
+			};
+		});
+
+		const blockKitParamters = {
+			date: Date.parse(latestRecord.date),
+			questionText: latestRecord.questionText,
+			responses,
+			answer: latestRecord.answer,
+			answerText: latestRecord.answerText
+		};
+
+		const blocks = latestRecord.answer
+			? blockKit.getClosedMessage(blockKitParamters)
+			: blockKit.getOpenMessage(blockKitParamters);
```

Comment:
Would it be helpful if we just merged these two functions and let the implementation of the common function figure out whether we're asking for an open or closed messages based on the parameters passed in? This would make your calling code a bit simpler.

Reply:
Yeah, that would make the calling code more simple. Also, the two methods will become better as well from this change.